library/src/main/java/io/appwrite/services/Database.kt

```diff
@@ -0,0 +1,197 @@
+package io.appwrite.services
+
+import android.net.Uri
+import io.appwrite.Client
+import io.appwrite.enums.OrderType
+import io.appwrite.exceptions.AppwriteException
+import okhttp3.Cookie
+import okhttp3.Response
+import java.io.File
+
+class Database(private val client: Client) : BaseService(client) {
+
+    /**
+     * List Documents
+     *
+     * Get a list of all the user documents. You can use the query params to
+     * filter your results. On admin mode, this endpoint will return a list of all
+     * of the project's documents. [Learn more about different API
+     * modes](/docs/admin).
+     *
+     * @param collectionId
+     * @param filters
+     * @param limit
+     * @param offset
+     * @param orderField
+     * @param orderType
+     * @param orderCast
+     * @param search
+     * @return The request response with a JSON body 
+     */
+    @JvmOverloads
+    @Throws(AppwriteException::class)
+    suspend fun listDocuments(
+		collectionId: String,
+		filters: List<Any>? = null,
+		limit: Int = 25,
+		offset: Int = 0,
+		orderField: String = "",
+		orderType: OrderType = OrderType.ASC,
+		orderCast: String = "string",
+		search: String = ""
+	): Response {
+        val path = "/database/collections/{collectionId}/documents".replace("{collectionId}", collectionId)
+        val params = mapOf<String, Any?>(
+            "filters" to filters,
+            "limit" to limit,
+            "offset" to offset,
+            "orderField" to orderField,
+            "orderType" to orderType.name,
+            "orderCast" to orderCast,
+            "search" to search
+        )
+
+        val headers = mapOf(
+            "content-type" to "application/json"
+        )
+
+        return client.call("GET", path, headers, params)
+    }
+    
+    /**
+     * Create Document
+     *
+     * Create a new Document. Before using this route, you should create a new
+     * collection resource using either a [server
+     * integration](/docs/server/database#databaseCreateCollection) API or
+     * directly from your database console.
+     *
+     * @param collectionId
+     * @param data
+     * @param read
+     * @param write
+     * @param parentDocument
+     * @param parentProperty
+     * @param parentPropertyType
+     * @return The request response with a JSON body 
+     */
+    @JvmOverloads
+    @Throws(AppwriteException::class)
+    suspend fun createDocument(
+		collectionId: String,
+		data: Any?,
+		read: List<Any>? = null,
+		write: List<Any>? = null,
+		parentDocument: String = "",
+		parentProperty: String = "",
+		parentPropertyType: String = "assign"
```

Comment:
We should avoid default values in the SDK level, and let the server handle them. Any optional param should have a null value, and if null the param shouldn't be send to the client.call method.

Reply:
Done :+1: